src/fs_sys.c

```diff
@@ -0,0 +1,342 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#include "fs_sys.h"
+#include "string.h"
+#include "kerneltypes.h"
+#include "process.h"
+#include "iso.h"
+#include "console.h"
+#include "fs_sys_err.h"
+
+#define READ 4
+#define WRITE 2
+#define APPEND 1
+
+uint32_t fs_sys_security_check(const char *path);
+
+/*
+ * A dummy function to help us in our debugging
+ */
+void fs_print_allowances() {
+    struct fs_allowance *iterator = current->fs_allowances_head;
+    while (iterator) {
+        console_printf("%s -> ", iterator->path);
+        iterator = iterator->next;
+    }
+    console_printf("\n");
+}
+
+void fs_sys_init_security(struct process *p) {
+    //Open files table
+    p->files = kmalloc(sizeof(*(p->files)));
+    int i;
+
+    //set all open file pointers to null
+    for (i = 0; i < PROCESS_MAX_OPEN_FILES; i++) {
+        p->files->open_files[i] = 0;
+    }
+
+    p->files->num_open = 0;
+
+    //Allowances alloc'ing.
+    p->fs_allowances_head = kmalloc(sizeof(*(p->fs_allowances_head)));
+
+    //Set the lone permission in the list to root (that is all of the fs)
+    p->fs_allowances_head->next = 0;
+    strcpy(p->fs_allowances_head->path, "/");
+    p->fs_allowances_head->do_allow_below = 1;
+}
+
+// Look at the OS's table of open files to see if read write conflicts
+// might be occurring amongst many processes if you allow it to be opened
+bool fs_sys_check_open_conflict(const char *path, const char *mode) {
+    return 1;
```

Comment:
We can put one here, but I'm also going to make a JIRA issue for having the OS keep an open files table. The problem is we can't really check it (or it will always return an okay) until we have a writable section of the fs which makes a conflict actually occur.